server/sonar-server/src/main/java/org/sonar/server/permission/ws/CreateTemplateAction.java

```diff
@@ -0,0 +1,109 @@
+import static org.sonar.server.permission.ws.Parameters.PARAM_TEMPLATE_NAME;
+import static org.sonar.server.permission.ws.Parameters.PARAM_TEMPLATE_PATTERN;
+import static org.sonar.server.permission.ws.Parameters.createTemplateDescriptionParameter;
+import static org.sonar.server.permission.ws.Parameters.createTemplateProjectKeyPatternParameter;
+import static org.sonar.server.permission.ws.PermissionTemplateDtoBuilder.newPermissionTemplateDto;
+import static org.sonar.server.permission.ws.PermissionTemplateDtoToPermissionTemplateResponse.toPermissionTemplateResponse;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+
+public class CreateTemplateAction implements PermissionsWsAction {
+  private final DbClient dbClient;
+  private final UserSession userSession;
+  private final System2 system;
+
+  public CreateTemplateAction(DbClient dbClient, UserSession userSession, System2 system) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.system = system;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction("create_template")
+      .setDescription("Create a permission template.<br />" +
+        "It requires administration permissions to access.")
+      .setResponseExample(getClass().getResource("create_template-example.json"))
+      .setSince("5.2")
+      .setPost(true)
+      .setHandler(this);
+
+    action.createParam(PARAM_TEMPLATE_NAME)
+      .setRequired(true)
+      .setDescription("Name")
+      .setExampleValue("Financial Service Permissions");
+
+    createTemplateProjectKeyPatternParameter(action);
+    createTemplateDescriptionParameter(action);
+  }
+
+  @Override
+  public void handle(Request wsRequest, Response wsResponse) throws Exception {
+    String name = wsRequest.mandatoryParam(PARAM_TEMPLATE_NAME);
+    String description = wsRequest.param(PARAM_TEMPLATE_DESCRIPTION);
+    String projectPattern = wsRequest.param(PARAM_TEMPLATE_PATTERN);
+
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      checkGlobalAdminUser(userSession);
+      validateTemplateNameForCreation(dbSession, dbClient, name);
+      validateProjectPattern(projectPattern);
+
+      PermissionTemplateDto permissionTemplate = insertTemplate(dbSession, name, description, projectPattern);
+
+      Permissions.CreatePermissionTemplateResponse response = buildResponse(permissionTemplate);
+      writeProtobuf(response, wsRequest, wsResponse);
+    } finally {
+      dbClient.closeSession(dbSession);
+    }
+  }
+
+  private PermissionTemplateDto insertTemplate(DbSession dbSession, String name, String description, String projectPattern) {
+    return dbClient.permissionTemplateDao().insert(dbSession, newPermissionTemplateDto(system)
+      .setName(name)
+      .setDescription(description)
+      .setProjectKeyPattern(projectPattern)
+      .toDto());
```

Comment:
because the commit is done in the DAO and shared with other methods. I'll add the commit when I'll remove other uses of the method